nemo/collections/nlp/data/dialogue/dataset/dialogue_bert_dataset.py

```diff
@@ -82,15 +86,150 @@ def __init__(self, dataset_split: str, dialogues_processor: object, tokenizer, c
             ignore_extra_tokens=self.cfg.ignore_extra_tokens,
             ignore_start_end=self.cfg.ignore_start_end,
         )
-
+        
         self.all_input_ids = features[0]
         self.all_segment_ids = features[1]
         self.all_input_mask = features[2]
         self.all_loss_mask = features[3]
         self.all_subtokens_mask = features[4]
         self.all_slots = features[5]
+
```

Comment:
A few general things: 
1. Write some inline comments for what the function is seeking to do what the expected inputs and outputs are? an illustrative example in the description will also be helpful if it's hard to understand without 
For instance
"""<overview of the function>
    Args:
        input_ids: input/token ids
        input_mask: masks out subword tokens
        segment_ids: distinguish one sentence from the other one (if present)
        label_ids: label for the current example
    """
2. For methods like yours below where the logic is very complex, please write some unit tests to make sure that how a function actually works is the same as how you expect it to work. See an example in https://github.com/NVIDIA/NeMo/blob/main/tests/collections/nlp/test_dialogue.py